src/Surfnet/Stepup/Identity/Identity.php

```diff
@@ -235,24 +238,74 @@ public function verifyEmail($verificationNonce)
         $secondFactorToVerify->verifyEmail();
     }
 
-    public function vetSecondFactor($registrationCode, $secondFactorIdentifier, $documentNumber, $identityVerified)
-    {
-        foreach ($this->verifiedSecondFactors as $secondFactor) {
-            if (!$secondFactor->wouldBeVettedBy(
-                $registrationCode,
-                $secondFactorIdentifier,
-                $documentNumber,
-                $identityVerified
-            )) {
-                continue;
-            }
+    public function vetSecondFactor(
+        IdentityApi $identity,
+        SecondFactorId $secondFactorId,
+        $secondFactorIdentifier,
+        $registrationCode,
+        $documentNumber,
+        $identityVerified
```

Comment:
Would be nice to wrap the last three params in a VO rather then keep passing em around. Also, I wonder if `identityVerified` should be passed down all the way to the SF, rather than do it here. Semantically it might be right to do as well: `If I have not verified the Identity, I may not vet this SecondFactor`